src/resources/js/admin/help-hub-iframe.js

```diff
@@ -76,125 +81,189 @@ window.DocsBotAI = window.DocsBotAI || {};
-			script.id = 'ze-snippet';
-			script.src = `https://static.zdassets.com/ekr/snippet.js?key=${ zendeskKey }`;
-			script.async = true;
-
-			document.head.appendChild( script );
-
-			script.onload = () => resolve();
-			script.onerror = () => reject( new Error( 'Failed to load Zendesk Web Widget' ) );
-		} );
+	obj.isOptedIn = function() {
+		var el = obj.getHelpHubPageElement();
+		return el && el.getAttribute('data-opted-in') === '1';
 	};
 
 	/**
-	 * Initializes the Zendesk widget, hides it initially, and sets up event listeners for open/close actions.
+	 * HelpScoutManager handles Help Scout Beacon integration.
 	 *
-	 * @link https://support.zendesk.com/hc/en-us/articles/4408836216218-Using-Web-Widget-Classic-to-embed-customer-service-in-your-website
+	 * @since TBD
+	 * @class
+	 * @link https://developer.helpscout.com/beacon-2/web/javascript-api
+	 */
+	function HelpScoutManager(beaconKey, userIdentifiers) {
+		this.beaconKey = beaconKey;
+		this.userIdentifiers = userIdentifiers || null;
+		this.scriptLoaded = false;
+		this.beaconReady = false;
+		this.initPromise = null;
+	}
+
+	/**
+	 * Loads the Help Scout Beacon script dynamically.
 	 *
-	 * @since 6.3.2
-	 * @return {void}
+	 * @since TBD
+	 * @return {Promise<void>}
 	 */
-	obj.initializeZendesk = () => {
-		obj.isZendeskInitialized = false;
-		const bodyElement = document.getElementById( obj.selectors.helpHubPageID );
+	HelpScoutManager.prototype.loadScript = function() {
+		const self = this;
+		if (self.scriptLoaded) return Promise.resolve();
 
-		zE( 'webWidget', 'hide', () => {
-			obj.isZendeskInitialized = true;
-		} );
+		// Replicate the official Help Scout Beacon stub logic (ES6)
+		if (!window.Beacon || !window.Beacon.readyQueue) {
+			window.Beacon = function(method, options, data) {
+				window.Beacon.readyQueue.push({ method, options, data });
+			};
+			window.Beacon.readyQueue = [];
+		}
 
-		// Add 'blackout' class when the widget is opened.
-		zE( 'webWidget:on', 'open', () => {
-			if ( obj.isZendeskInitialized ) {
-				bodyElement.classList.add( 'blackout' );
-			}
-		} );
+		return new Promise((resolve, reject) => {
+			const script = document.createElement('script');
+			script.src = 'https://beacon-v2.helpscout.net';
+			script.async = true;
+			script.onload = () => {
+				self.scriptLoaded = true;
+				resolve();
+			};
+			script.onerror = () => {
+				reject(new Error('Failed to load Help Scout Beacon script.'));
+			};
+			// Insert into <head> for best compatibility
+			document.head.appendChild(script);
+		});
+	};
 
-		// Remove 'blackout' class when the widget is closed.
-		zE( 'webWidget:on', 'close', () => {
-			zE( 'webWidget', 'hide' );
-			bodyElement.classList.remove( 'blackout' );
-		} );
+	/**
+	 * Initializes the Help Scout Beacon widget.
+	 *
+	 * @since TBD
+	 * @return {Promise<void>}
+	 */
+	HelpScoutManager.prototype.initBeacon = function() {
+		const self = this;
+		if (self.initPromise) return self.initPromise;
```

Comment:
Always use braces.

Also, it would be nice to have a more specific conditional check than just whether it is truthy.

```suggestion
		if ( self.initPromise ) {
			return self.initPromise;
		}
```